rotkehlchen/icons.py

```diff
@@ -127,26 +130,31 @@ def get_icon(
                 image_data = f.read()
             return image_data
 
-        # Then our only chance is coingecko
-        try:
-            asset = asset.resolve_to_asset_with_oracles()
-        except (UnknownAsset, WrongAssetType):
-            return None
+        # check if the asset is in a collection
+        related_assets_identifiers = set(GlobalDBHandler().get_related_assets(asset.identifier))
+
+        asset_to_query_icon = asset
+        if len(related_assets_identifiers) != 0:
+            related_assets_identifiers.add(asset.identifier)
+            # get the asset with the lowest lex order
+            collection_main_asset_id = min(related_assets_identifiers)
```

Comment:
Why no to do these checks at the DB query time?

Something like method `get_identifier_with_icon(identifier: str)` that would accept an identifier and return the asset identifier to check / query the icon for

Reply:
I made the method general enough to be useful in other places but I can return only the min I guess